src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/FunctionsMiddleware/FunctionsMcpToolResultMiddleware.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Middleware;
+// Need to do this in a way without bringing in the MCP SDK directly into this project.
+// This should be part of the wrapper project (wip)
+using ModelContextProtocol.Protocol;
```

Comment:
This is a draft because I think this kind of change needs to go into the wrapper we discussed for brining in the MCP SDK